libs/crypto/benchmark/bls_signing_and_verification.cpp

```diff
@@ -0,0 +1,141 @@
+  for (std::size_t i = 0; i < NUM_WORDS; ++i)
+  {
+    *words++ = rng();
+  }
+
+  return ConstByteArray{buffer};
+}
+
+void SignBLSSignature(benchmark::State &state)
+{
+  bn::initPairing();
+  // Generate a random message
+  ConstByteArray msg = GenerateRandomData<256>();
+
+  // Create keys
+  auto     committee_size = static_cast<uint32_t>(state.range(0));
+  uint32_t threshold      = committee_size / 2 + 1;
+  auto     outputs = fetch::crypto::mcl::TrustedDealerGenerateKeys(committee_size, threshold);
+  assert(outputs.size() == committee_size);
+
+  // Randomly select index to sign
+  auto index = static_cast<uint32_t>(rng() % committee_size);
+
+  for (auto _ : state)
+  {
+    // Sign message
+    fetch::crypto::mcl::SignShare(msg, outputs[index].private_key_share);
+  }
+}
+
+void VerifyBLSSignature(benchmark::State &state)
+{
+  bn::initPairing();
+  bn::G2 generator;
+  fetch::crypto::mcl::SetGenerator(generator);
+  // Generate a random message
+  ConstByteArray msg = GenerateRandomData<256>();
+
+  // Create keys
+  auto     committee_size = static_cast<uint32_t>(state.range(0));
+  uint32_t threshold      = committee_size / 2 + 1;
+  auto     outputs = fetch::crypto::mcl::TrustedDealerGenerateKeys(committee_size, threshold);
+
+  // Randomly select index to sign
+  auto sign_index = static_cast<uint32_t>(rng() % committee_size);
+  auto signature  = fetch::crypto::mcl::SignShare(msg, outputs[sign_index].private_key_share);
+
+  // Randomly select another index to verify
+  auto verify_index = static_cast<uint32_t>(rng() % committee_size);
+  bool check;
+  FETCH_UNUSED(check);
+  for (auto _ : state)
+  {
+    // Verify message
+    check = fetch::crypto::mcl::VerifySign(outputs[verify_index].public_key_shares[sign_index], msg,
+                                           signature, generator);
+  }
+  assert(check);
+}
+
+void ComputeGroupSignature(benchmark::State &state)
+{
+  bn::initPairing();
+  // Generate a random message
+  ConstByteArray msg = GenerateRandomData<256>();
+
+  // Create keys
+  auto     committee_size = static_cast<uint32_t>(state.range(0));
+  uint32_t threshold      = committee_size / 2 + 1;
+  auto     outputs = fetch::crypto::mcl::TrustedDealerGenerateKeys(committee_size, threshold);
+
+  // Randomly select indices to sign
+  std::unordered_map<uint32_t, fetch::crypto::mcl::Signature> threshold_signatures;
+  while (threshold_signatures.size() < threshold)
+  {
+    auto sign_index = static_cast<uint32_t>(rng() % committee_size);
+    auto signature  = fetch::crypto::mcl::SignShare(msg, outputs[sign_index].private_key_share);
+    threshold_signatures.insert({sign_index, signature});
+  }
+
+  fetch::crypto::mcl::Signature group_sig;
+  FETCH_UNUSED(group_sig);
+  for (auto _ : state)
+  {
+    // Compute group signature
+    group_sig = fetch::crypto::mcl::LagrangeInterpolation(threshold_signatures);
+  }
+  auto index = static_cast<uint32_t>(rng() % committee_size);
+  FETCH_UNUSED(index);
```

Comment:
You don't usually test anything in a benchmark - in this case devs would be unlikely to see this as you don't usually run the benchmark in debug mode where this assert would be enabled